pkg/artifactory/resource_artifactory_remote_cargo_repository.go

```diff
@@ -18,12 +18,19 @@ var cargoRemoteSchema = mergeSchema(baseRemoteSchema, map[string]*schema.Schema{
 		Description: "(On the UI: Anonymous download and search) Cargo client does not send credentials when performing download and search for crates. " +
 			"Enable this to allow anonymous access to these resources (only), note that this will override the security anonymous access option.",
 	},
+	"list_remote_folder_items": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: `(Optional) Lists the items of remote folders in simple and list browsing. The remote content is cached according to the value of the 'Retrieval Cache Period'. Default value is 'false'.`,
+	},
 })
 
 type CargoRemoteRepo struct {
 	RemoteRepositoryBaseParams
-	RegistryUrl     string `hcl:"git_registry_url" json:"gitRegistryUrl"`
-	AnonymousAccess bool   `hcl:"anonymous_access" json:"cargoAnonymousAccess"`
+	RegistryUrl           string `hcl:"git_registry_url" json:"gitRegistryUrl"`
+	AnonymousAccess       bool   `hcl:"anonymous_access" json:"cargoAnonymousAccess"`
+	ListRemoteFolderItems bool   `json:"listRemoteFolderItems"`
```

Comment:
Why not add this to `RemoteRepositoryBaseParams`?

Reply:
MessyRemoteRepo uses RemoteRepositoryBaseParams ... so will get this attribute too. Is it ok?

Reply:
You're right @maheshjfrog - As much as I don't want to add a feature to the legacy stuff, I also don't like C&P - Let's do this: 
😢  Literally C&P all the attributes of `RemoteRepositoryBaseParams` paste them directly into `MessyRemoteRepo`. This way, when it comes time to remove `Messy*` you delete the struct and it's just gone - I don't want to add any new features to the old code as this will discourage people from moving. 

So, to be clear:
1. Add all fields current in `BaseRemoteRepoParams` to `MessyRemoteRepo` 
2. Add the field to `BaseRemoteRepoParams`

Then you can do as alex recommends and remove it from each specific repo directly. Just make sure you add this field to `unpackBaseRemoteRepo`

Reply:
done. Moved BaseRemoteRepoParams out of MessyRemoteRepo. And, add listRemoteFolderItems to BaseRemoteRepoParams.